src/console/components/cz-console-2/main.js

```diff
@@ -0,0 +1,338 @@
+/**
+ * ConsoleComponent
+ */
+class ConsoleComponent extends HTMLElement {
+  /**
+   * Constructor
+   */
+  constructor() {
+    // Always call super first in constructor
+    super();
+
+    const shadow = this.attachShadow({mode: 'open'});
+
+    this.getAsert(shadow);
+    this.getClear(shadow);
+    this.getCount(shadow);
+    this.getcountReset(shadow);
+    this.getDebug(shadow);
+    this.getDir(shadow);
+    this.getDirXML(shadow);
+    this.getError(shadow);
+    this.getGroup(shadow);
+    this.getGroupCollapsed(shadow);
+    this.getInfo(shadow);
+    this.getLog(shadow);
+    this.getProfile(shadow);
+    this.getProfileEnd(shadow);
+    this.getTable(shadow);
+    this.getTime(shadow);
+    this.getTimeEnd(shadow);
+    this.getTimeLog(shadow);
+    this.getTimeStamp(shadow);
+    this.getTrace(shadow);
+    this.getWarn(shadow);
+  }
+  /**
+   *
+   * @param {shadowroot} shadow
+   */
+  getAsert(shadow) {
+    const button = document.createElement('button');
+    button.textContent = 'Assert';
+    button.onclick = () => {
+      const number = Math.floor(Math.random() * 10);
+      console.assert(number % 2 === 0,
+        {errorMsg: `Number ${number} is not even`});
+      console.log(`Number ${number}`);
+    };
+    shadow.appendChild(button);
+  }
+
+  /**
+   *
+   * @param {shadowroot} shadow
+   */
+  getClear(shadow) {
+    const button = document.createElement('button');
+    button.textContent = 'Clear';
+    button.onclick = () => {
+      console.clear();
+    };
+    shadow.appendChild(button);
+  }
+
+  /**
+   *
+   * @param {shadowroot} shadow
+   */
+  getCount(shadow) {
+    const button = document.createElement('button');
+    button.textContent = 'Count';
+    button.onclick = () => {
+      console.count();
+    };
+    shadow.appendChild(button);
+  }
+
+  /**
+   *
+   * @param {shadowroot} shadow
+   */
+  getcountReset(shadow) {
+    const button = document.createElement('button');
+    button.textContent = 'countReset';
+    button.onclick = () => {
+      console.countReset();
+    };
+    shadow.appendChild(button);
+  }
+
+  /**
+   *
+   * @param {shadowroot} shadow
+   */
+  getDebug(shadow) {
+    const button = document.createElement('button');
+    button.textContent = 'Debug';
+    button.onclick = () => {
+      console.debug('Debugging');
+    };
+    shadow.appendChild(button);
+  }
+  /**
+   *
+   * @param {shadowroot} shadow
+   */
+  getDir(shadow) {
+    const button = document.createElement('button');
+    button.textContent = 'Dir';
+    button.onclick = () => {
+      console.dir(document.location);
+    };
+    shadow.appendChild(button);
+  }
+  /**
+   *
+   * @param {shadowroot} shadow
+   */
+  getDirXML(shadow) {
+    const button = document.createElement('button');
+    button.textContent = 'DirXML';
+    button.onclick = () => {
+      console.dirxml(document.location);
+    };
+    shadow.appendChild(button);
```

Comment:
It would be nice to change the element we print in the line 123 to exemplify better how console.dirxml displays an XML/HTML representation when possible